pkg/controllers/nodeclass/status/images.go

```diff
@@ -147,9 +171,50 @@ func imageVersionsUnready(nodeClass *v1alpha2.AKSNodeClass) bool {
 }
 
 // Handles case 4: check if the maintenance window is open
-func isMaintenanceWindowOpen() bool {
-	// TODO: need to add the actual logic for handling maintenance windows once it is in the ConfigMap.
-	return true
+func (r *NodeImageReconciler) isMaintenanceWindowOpen(ctx context.Context) (bool, error) {
+	if !r.aksControlPlane {
+		// Note: when aksControlPlane is false, continuous upgrade is enabled without checking the maintenane windows.
+		return true, nil
+	}
+	systemNamespace := strings.TrimSpace(os.Getenv("SYSTEM_NAMESPACE"))
```

Comment:
Is pulling this directly from the Env here best, or should we do it at an earlier point/in a different way?

Reply:
I do somewhat prefer doing it here, as if it is missing from the env we still get come update paths, and will only fail when checking the maintenance windows.

Reply:
Reading from environment on every call seems unnecesary. Whatever logic you need when it is not set can still be used if it is initialized once.

Reply:
Good callout. Moved to the constructor, and simply preforms the same logic with the stored value, instead of getting it from the env each time.